### ISSUE TITLE
Added handling for ENOMEM errors in `bittorrent-lsd`

### ISSUE THREAD

**Issue by feross** [webtorrent/webtorrent#1956]
I just attempted to release 0.111.0 but I'm blocked on an error in the tests:

```
# client.seed followed by duplicate client.add (async)
dgram.js:828
    throw errnoException(err, 'addMembership');
    ^

Error: addMembership ENOMEM
    at Socket.addMembership (dgram.js:828:11)
    at Socket.onListening (/Users/feross/code/webtorrent/node_modules/bittorrent-lsd/index.js:40:19)
    at Socket.emit (events.js:327:22)
    at startListening (dgram.js:149:10)
    at dgram.js:344:7
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  errno: -12,
  code: 'ENOMEM',
  syscall: 'addMembership'
```

It seems that calling `addMembership` in https://github.com/webtorrent/bittorrent-lsd can fail sometimes and we need to handle these errors.

I found this commit from another project which shows one possible way to handle the error, with retries: https://github.com/thaliproject/node-ssdp/commit/fcc5d403f6cc5e667572fba059332908c3cc822e We don't necessarily need something as complicated as retries, but we should at least capture the error.

@hicom150 If you have a second, could you take a look at this?

**Comment by hicom150** [comment]
@feross Made a quick PR to handle the error 😉

**Comment by feross** [comment]
@hicom150 Looks good. At least the error is handled now! Merged it.

But I think we probably want to actually emit that as a `warning` since the job of the package is to find peers and it's totally fine if it finds 0 peers -- that's not really an error. If we emit an error than the whole torrent object will be destroyed.

https://github.com/webtorrent/webtorrent/blob/e5445be2f5d64bc545211c604e106dc3bcb10385/lib/torrent.js#L340-L342

**Comment by hicom150** [comment]
Fixed in https://github.com/webtorrent/bittorrent-lsd/pull/5 and released as [bittorrent-lsd@v1.0.1](https://github.com/webtorrent/bittorrent-lsd/releases/tag/v1.0.1)

**Comment by feross** [comment]
Just released 0.111.0.
